lib/internal/watch_mode/files_watcher.js

```diff
@@ -98,7 +98,7 @@ class FilesWatcher extends EventEmitter {
     }
     const watcher = watch(path, { recursive, signal: this.#signal });
     watcher.on('change', (eventType, fileName) => this
-      .#onChange(recursive ? resolve(path, fileName) : path));
+      .#onChange(recursive ? resolve(path, fileName ?? '') : path));
```

Comment:
Can we add a comment to here referencing/mentioning @bnoordhuis's comments? It's important to not lose any knowledge we have.